agent/agentcontainers/containers_dockercli.go

```diff
@@ -183,7 +184,7 @@ func devcontainerEnv(ctx context.Context, execer agentexec.Execer, container str
 		return nil, nil
 	}
 
-	meta := make([]DevContainerMeta, 0)
+	meta := make([]dcspec.DevContainer, 0)
 	if err := json.Unmarshal([]byte(rawMeta), &meta); err != nil {
```

Comment:
Suggestion: We could first attempt a strict unmarshal via `dec := json.NewDecoder(), dec.DisallowUnknownFields()`, if it catches an issue we can log it and fall-back to non-strict unmarshaling.

Not strictly necessary, but may help provide hints down the line why something isn't working.